frontend/model/contracts/chatroom.js

```diff
@@ -426,7 +438,7 @@ sbp('chelonia/defineContract', {
           state.messages.push(createMessage({ meta, data, hash, height, state, pending, innerSigningContractID }))
         } else if (direction !== 'outgoing') {
           // If an existing message is found, it's no longer pending.
-          delete existingMsg.pending
+          existingMsg.pending = false
```

Comment:
This doesn't look very important, but why set it to `false`? Although a boolean probably doesn't cost much, it's still slightly more information to keep in memory.

Reply:
To keep the consistency. When the user receives the message sent by another, the `pending` would be `false`.
```
if (!existingMsg) {
  // If no existing message, simply add it to the messages array.
  const pending = direction === 'outgoing'
  state.messages.push(createMessage({ meta, data, hash, height, state, pending, innerSigningContractID }))
} else if (direction !== 'outgoing') {
  // If an existing message is found, it's no longer pending.
  existingMsg.pending = false
}
```

Reply:
I do agree with @corrideat here as both approaches result in the same value for `!existingMsg.pending` and `v-if='existingMsg.pending'`, but the `delete` approach uses slightly less memory